src/lib/storage/aws.ts

```diff
@@ -0,0 +1,57 @@
+import { S3Client, GetObjectCommand, PutObjectCommand } from '@aws-sdk/client-s3';
+import log from '../../helpers/log';
+import type { IStorage } from './types';
+
+class Aws implements IStorage {
+  client: S3Client;
+  folder: string;
+
+  constructor(folder: string) {
+    const region = process.env.AWS_REGION;
+    const accessKeyId = process.env.AWS_ACCESS_KEY_ID;
+    const secretAccessKey = process.env.AWS_SECRET_ACCESS_KEY;
+
+    if (!region || !accessKeyId || !secretAccessKey) {
+      throw new Error('[storage:aws] AWS credentials missing');
+    }
```

Comment:
We are not using these anywhere? will it work? 🤔

Reply:
Yes, AWS SDK is picking up the env variable transparently, when using the next line `this.client = new S3Client({});`